data/src/main/java/com/microsoft/azure/kusto/data/http/HttpClientFactory.java

```diff
@@ -24,68 +21,49 @@ public class HttpClientFactory {
     private static final Logger LOGGER = LoggerFactory.getLogger(HttpClientFactory.class);
 
     /**
-     * Creates a new Apache HTTP client.
+     * Creates a new HTTP client.
      *
      * @param providedProperties custom HTTP client properties
-     * @return a new Apache HTTP client
+     * @return a new HTTP client
      */
-    public static CloseableHttpClient create(HttpClientProperties providedProperties) {
-        LOGGER.info("Creating new CloseableHttpClient client");
+    public static HttpClient create(HttpClientProperties providedProperties) {
```

Comment:
The code in httpclientprovider can re-use the client if it has "null" as clientProperties.

Does it make sense to send null to the builder if providedProperties is null?

Reply:
@AsafMah I just checked and I carried this over from 5.0.5 so the behavior is the same as in v5.

Currently, if we send null to the builder, then the builder returns some sensible default values for most properties. I think that makes sense but I also think the HttpClient in azure-core does the same thing. If you and your team want to control the defaults then it makes sense to call the builder and have it return its default values. If you'd rather let azure-core decide the defaults, then it makes sense to skip the builder and just go straight to returning a client with no properties.

Reply:
@AsafMah I submitted a commit for this that just lets azure-core decide defaults. This makes us less dependent on the builder in this codebase which makes us less dependent on our wrapper. We may want to run the JMeter tests again to see what those defaults do to response times etc.
[Commit is here.](https://github.com/Azure/azure-kusto-java/pull/335/commits/5d3a6ed12e2ce53c7f3221910168ad4182b67c00)